api/src/main/java/org/openmrs/validator/AllergyValidator.java

```diff
@@ -64,6 +64,11 @@ public void validate(Object target, Errors errors) {
 		
 		Allergy allergy = (Allergy) target;
 		
+		    // Additional validation: Ensure allergen does not contain numeric values
+			if (allergy.getAllergen() != null && StringUtils.isNumeric(allergy.getAllergen().getNonCodedAllergen())) {
+				errors.rejectValue("allergen", "error.allergyapi.allergy.Allergen.cannotContainNumeric");
+			}
+			
 		if (allergy.getReactionNonCoded() != null) {
```

Comment:
Do the same here

Reply:
`if (Objects.nonNull(allergy.getReactionNonCoded()))` is this is OK?

Reply:
try to do as described above or follow the the [Java Conventions](https://wiki.openmrs.org/display/docs/Java+Conventions)